src/main/java/hudson/plugins/build_timeout/BuildStepWithTimeout.java

```diff
@@ -0,0 +1,153 @@
+package hudson.plugins.build_timeout;
+
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.model.*;
+import hudson.plugins.build_timeout.impl.AbsoluteTimeOutStrategy;
+import hudson.plugins.build_timeout.impl.DeadlineTimeOutStrategy;
+import hudson.plugins.build_timeout.operations.AbortOperation;
+import hudson.tasks.BuildStep;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.BuildStepMonitor;
+import hudson.tasks.Builder;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+import org.jenkinsci.plugins.tokenmacro.MacroEvaluationException;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+
+public class BuildStepWithTimeout extends Builder implements BuildStep {
+    private /* final */ BuildTimeOutStrategy strategy;
+    private BuildStep buildStep;
```

Comment:
Would you consider to accept multiple steps?
That should make this feature more useful.

You can do that by using `<f:hetero-list>` instead of `<f:dropdownDescriptor>`.
This review may be helpful: https://github.com/jenkinsci/flexible-publish-plugin/pull/6/files